### PR TITLE
Update backlog page

### PR DESCRIPTION
**Pull request by christopherthomasdesign**
Update to the community backlog page. Along with some changes to the Community Backlog Github board, it solves #1958 

The main changes are:
* better introducing the backlog and what to use it for
* removing the list of backlog items, so that the Github board is the single source of truth

### CODE REVIEW

**Comment by calvin-lau-sig7** [comment]
@christopherthomasdesign Okay, I've added changes to the pages that reference the status of contributions in the backlog.

I ran into some trouble with the commit history whilst squashing, but I think I've restored things. Appreciate if you can take a look!

File: src/community/contribution-criteria/index.md.njk [hunk 1]
```diff
@@ -52,7 +52,7 @@ To be successful, proposals need to show that the component or pattern being sug
 }) }}
 
 
-The [Design System working group](/community/design-system-working-group/) reviews proposals in the [community backlog](/community/backlog/) to check they meet these criteria. Proposals that meet the criteria will then be marked 'to do', ready to be worked on.
+The [Design System working group](/community/design-system-working-group/) reviews proposals in the [community backlog](/community/backlog/) to check they meet these criteria. Proposals that meet the criteria will then moved to the next step, 'Needs more detail', to show they're ready to be worked on.
```
Comment (EoinShaughnessy) [inline src/community/contribution-criteria/index.md.njk:55]:
```suggestion
The [Design System working group](/community/design-system-working-group/) reviews proposals in the [community backlog](/community/backlog/) to check they meet these criteria. Proposals that meet the criteria then move to the next step, 'Needs more detail', to show they're ready to be worked on.
```

File: src/community/propose-a-component-or-pattern/index.md.njk [hunk 2]
```diff
@@ -12,17 +12,17 @@ To be successful, proposals need to show that the component or pattern being sug
...
 Follow the 3 steps to propose a component or pattern for the Design System.
 
-## 1. Check the community backlog
+## 1. Check the Community Backlog
 
-Check if someone else has already suggested your idea or something similar on the [community backlog](/community/backlog/). 
+Check if someone else has already suggested your idea or something similar on the [Community Backlog project board on GitHub](https://github.com/alphagov/govuk-design-system-backlog/projects/1). 
 
-If your idea is on the backlog and marked as 'proposed', follow the link and comment on the issue. Say you need the component or pattern and share any examples or evidence you have to support the proposal.
+If your idea is on the backlog and marked as 'Needs more evidence', go into the issue page and comment on the issue. Say you need the component or pattern and share any examples or evidence you have to support the proposal.
```
Comment (EoinShaughnessy) [inline src/community/propose-a-component-or-pattern/index.md.njk:19]:
```suggestion
If your idea is on the backlog and marked as 'Needs more evidence', go into the issue page and comment on the issue. Say you need the component or pattern, and share any examples or evidence you have to support the proposal.
```

File: src/community/propose-a-component-or-pattern/index.md.njk [hunk 3]
```diff
@@ -12,17 +12,17 @@ To be successful, proposals need to show that the component or pattern being sug
...
 
-If it’s marked 'to do' and you would like to work on it, read how to [develop a component or pattern](/community/develop-a-component-or-pattern/).
+If it’s marked as 'Needs more detail' and you'd like to work on it, read how to [develop a component or pattern](/community/develop-a-component-or-pattern/).
 
 ## 2. Raise an issue
 
-If your idea is not on the backlog, [raise an issue using the template](https://github.com/alphagov/govuk-design-system-backlog/issues/new). A member of the Design System team will get in touch to discuss your proposal. 
+If your idea is not on the backlog, [raise an issue using the template](https://github.com/alphagov/govuk-design-system-backlog/issues/new). A member of the Design System team will get in touch to discuss your proposal.
```
Comment (EoinShaughnessy) [inline src/community/propose-a-component-or-pattern/index.md.njk:25]:
```suggestion
If your idea is not on the backlog, [raise an issue](https://github.com/alphagov/govuk-design-system-backlog/issues/new). A member of the Design System team will get in touch to discuss your proposal.
```

File: src/community/propose-a-component-or-pattern/index.md.njk [hunk 4]
```diff
@@ -36,9 +36,6 @@ At the meeting, the working group will review your proposal and decide if it is
 
 After the meeting, the Design System team will let you know the decision and recommendations, if there are any.
 
-If the working group agrees your proposal is useful and unique, the Design System team will mark it as 'to do' on the [community backlog](/community/backlog/). 
+If the working group agrees your proposal is useful and unique, the Design System team will move it to the next step and mark it as 'Needs more detail' on the [Community Backlog project board on GitHub](https://github.com/alphagov/govuk-design-system-backlog/projects/1).
```
Comment (EoinShaughnessy) [inline src/community/propose-a-component-or-pattern/index.md.njk:39]: I think GOV.UK convention is for sentences to be no longer than 25 words.

```suggestion
If the working group agrees your proposal is useful and unique, the Design System team will:

- move the proposal to the next step
- mark it as 'Needs more detail' on the [Community Backlog project board on GitHub](https://github.com/alphagov/govuk-design-system-backlog/projects/1)
```

Bonus question: are "next step" and "'Needs more detail'" the same thing?

**Review by EoinShaughnessy — COMMENTED** [review]
Updates are looking great! Added a few comments.

**Comment by calvin-lau-sig7** [comment]
Thanks for the comments @EoinShaughnessy, I've committed them. 
Can I get another review and approval if it looks good? 😁

**Review by EoinShaughnessy — APPROVED** [review]

**Pull request merged**